### ISSUE TITLE
Diagnostics not working if first source file is on a different drive than editor application

### ISSUE THREAD

**Issue by Ekopalypse** [clojure-lsp/clojure-lsp#1042]
I'm not sure if this is either me doing something wrong with my lsp client for Noteapd++ (Npp), or whether this is a clojure-lsp or a clj-kondo issue.

If the initial source file used to start the lsp server is on a drive other than Npp, it does not work. However, if it is on the same drive, source files from other drives can be opened afterwards. 

![image](https://user-images.githubusercontent.com/47723516/173190445-03ace2bf-a1d7-435e-b665-3341ba9636a3.png)

I did some testing and the pattern got confirmed

~~~
Npp started from C:
	LSP executable on C:
		initial source is on C:	ok
		initial source is on D:	fail
	LSP executable on D:
		initial source is on C:	ok
		initial source is on D:	fail
Npp started from D:
	LSP executable on C:
		initial source is on C:	fail
		initial source is on D:	ok
	LSP executable on D:
		initial source is on C:	fail
		initial source is on D:	ok
~~~

I have tested the same with gopls (Go lsp server) and do not have this problem there,
which leads me to think that this is somehow related to Clojure. What do you think?

**Comment by ericdallo** [comment]
Seems like the exception is coming from clj-kondo, are you aware of any issue like that @borkdude ?
I don't have a Windows to manually test that unfortunately

**Comment by borkdude** [comment]
Perhaps you can test with clj-kondo only?

**Comment by Ekopalypse** [comment]
I can try, can you give me some hints on what I need to do? I'm not a Clojure user yet.

**Comment by ericdallo** [comment]
@Ekopalypse if you install [latest clj-kondo](https://github.com/clj-kondo/clj-kondo/blob/master/doc/install.md) and then run `clj-kondo --lint C:\x64\test_clojure\main.clj` should be enough I guess

**Comment by Ekopalypse** [comment]
@ericdallo - Thx - I'll give it a try.

**Comment by Ekopalypse** [comment]
Yes, this seems to work. I called it directly from a cmd prompt and from within Notepad++.

**Comment by Ekopalypse** [comment]
@ericdallo - Just found out that the clojure-lsp has a diagnostic command option.
Is calling `clojure-lsp.exe diagnostics --filenames C:\x64\test_clojure\main.clj` doing the same
as with didOpen notification?

**Comment by ericdallo** [comment]
it's similar @Ekopalypse , but I think the issue is related with clj-kondo --copy-configs args that clojure-lsp pass it as default.
Try calling kondo with that copy-configs arg
[ref](https://github.com/clojure-lsp/clojure-lsp/blob/master/lib/src/clojure_lsp/kondo.clj#L230)

**Comment by Ekopalypse** [comment]
I ran `C:\Temp\clojure\clj-kondo.exe --copy-configs --lint D:\x64\test_clojure\main.clj`
and it did not throw an exception.

**Comment by ericdallo** [comment]
try removing `.clj-kondo/.cache`

**Comment by ericdallo** [comment]
BTW The exception is thrown [here](https://github.com/clj-kondo/clj-kondo/blob/master/src/clj_kondo/impl/core.clj#L504)

**Comment by ericdallo** [comment]
@Ekopalypse could you paste the server info log? it's a useful clojure-lsp log that logs lots of information about the server, check [here](https://clojure-lsp.io/capabilities/#server-info-log), it's a LSP command, client needs to call

**Comment by Ekopalypse** [comment]
@ericdallo - Sorry, I am currently busy with a business matter. I'll get back to you as soon as I get the issue resolved. Thank you very much with all of your help so far.

**Comment by Ekopalypse** [comment]
@ericdallo - I've create a [gist](https://gist.github.com/Ekopalypse/30184e2dafd1d261d6940344c34943cb) with all the information requested. 
Running clj-kondo.exe from cmd does not create a ./clj-kondo directory within the source file directory nor within the current working directory. Is this expected?

By the way, how to implement the clojure/serverInfo/log method correctly?
I tried

~~~json
    {"jsonrpc":"2.0","id":2,"method":"clojure/serverInfo/log","params":{}}
    {"jsonrpc":"2.0","id":2,"method":"clojure/serverInfo/log"}
    {"jsonrpc":"2.0","id":2,"method":"clojure/serverInfo/log","params":null}
~~~

but all produced an
`org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer fireError
SEVERE: An error occurred while processing an incoming message.
java.lang.NullPointerException` exception

however the lsp server has returned something.

Have a nice sunday.

**Comment by ericdallo** [comment]
> Running clj-kondo.exe from cmd does not create a ./clj-kondo directory within the source file directory nor within the current working directory. Is this expected?

Yes, clojure-lsp creates it automatically, you can create it manually so.

I'll check the proper way to request the server info logs

**Comment by Ekopalypse** [comment]
Tested with [this version](https://github.com/clojure-lsp/clojure-lsp/releases/tag/2025.11.28-12.47.43), problem fixed.
